packages/webdriverio/src/commands/element/isDisplayed.js

```diff
@@ -34,20 +34,47 @@
  * </example>
  *
  * @alias element.isDisplayed
- * @return {Boolean} true if element(s)* [is|are] displayed
+ * @return {Boolean} true if element is displayed
  * @uses protocol/elements, protocol/elementIdDisplayed
  * @type state
  *
  */
 
+import { ELEMENT_KEY } from '../../constants'
+import { getBrowserObject } from '../../utils'
+import isElementDisplayedScript from '../../scripts/isElementDisplayed'
+
 export default async function isDisplayed() {
 
+    let browser = getBrowserObject(this)
+
     /*
      * This is only necessary as isDisplayed is on the exclusion list for the middleware
      */
     if (!this.elementId) {
         this.elementId = (await this.parent.$(this.selector)).elementId
     }
 
-    return this.elementId ? await this.isElementDisplayed(this.elementId) : false
+    /*
+     * if element was still not found it also is not displayed
+    */
+    if (!this.elementId) {
+        return false
+    }
+
+
+    //Default is to use the endpoint
+    let useEndpoint = true
+
+    //Safari >= 12 and Edge 18
+    if (browser.isW3C && ['MicrosoftEdge', 'safari'].includes(browser.capabilities.browserName)) {
```

Comment:
```suggestion
    if (browser.isW3C && browser.capabilities.browserName && ['microsoftedge', 'safari'].includes(browser.capabilities.browserName.toLowerCase())) {
```

also we can move `['microsoftedge', 'safari']` as constant to the top of the file.

Reply:
Done.